### PR TITLE
288 merge from master

### PR DESCRIPTION
**Pull request by cmarooney-stfc**
Merge master into the new sqw rewrite feature branch 288_sqw_rewrite feature, and then push to the feature branch.

Comments %CMDEV_MERGE show any issues with the choice of confict

### CODE REVIEW

File: horace_core/@d3d_old/cut.m [hunk 1]
```diff
@@ -52,14 +52,21 @@
 
 % Perform operations
 % ------------------
+if iscell(w.data)&& ischar(w.data{1})
+    % Slightly hacky - load dnd object before passing it to sqw.cut so we don't
+    % attempt to load the file in as an SQW file. We reach this point if cut_dnd
+    % is called.
+    % After DnDBase class is implemented, cutting a DnD object need no longer
+    % go through sqw.cut and this file will be largely re-written
+    w.data = d3d(w.data{1});
```
Comment (nickbattam-tessella) [inline horace_core/@d3d_old/cut.m:61]: `d3d_old`? (same for the other repetitions of this code fragment)

Comment (cmarooney-stfc) [reply horace_core/@d3d_old/cut.m:61]: @nickbattam-tessella This was automatically merged from master when it was d3d. I can fix this (and similar) back to `_old` , thanks for spotting

File: horace_core/sqw/@DnDBase/DnDBase.m [hunk 2]
```diff
@@ -55,6 +55,12 @@
         function obj = DnDBase(varargin)
             obj = obj@SQWDnDBase();
 
+			% conversion to struct done here rather than 
+			% in parse_args_ as the need for this conversion
+			% may be temporary
+            if nargin>0 && isa(varargin{1},'data_sqw_dnd')
```
Comment (nickbattam-tessella) [inline horace_core/sqw/@DnDBase/DnDBase.m:61]: this logic should be in parse_args

**Review by nickbattam-tessella — COMMENTED** [review]
Updates to make the tests pass have to be done as soon as possible; needing to disable the tests means the merge hasn't been completed successfully.

**Pull request merged**